ereuse_devicehub/resources/action/schemas.py

```diff
@@ -64,21 +67,55 @@ class Remove(ActionWithOneDevice):
 
 class Allocate(ActionWithMultipleDevices):
     __doc__ = m.Allocate.__doc__
-    to = NestedOn(s_user.User,
-                  description='The user the devices are allocated to.')
-    organization = SanitizedStr(validate=Length(max=STR_SIZE),
-                                description='The organization where the '
-                                            'user was when this happened.')
+    start_time = DateTime(data_key='startTime', required=True,
+                          description=m.Action.start_time.comment)
+    end_time = DateTime(data_key='endTime', required=False,
+                        description=m.Action.end_time.comment)
+    code = SanitizedStr(data_key='transaction', validate=Length(min=1, max=STR_BIG_SIZE),
+                        required=False,
+                        description='The code of the agent to assigned.')
+    end_users = Integer(data_key='endUsers', validate=[Range(min=1, error="Value must be greater than 0")])
```

Comment:
Check if the end_user variable is a required=True attribute within the Allocate action.